crates/agglayer-config/src/prover.rs

```diff
@@ -67,21 +64,47 @@ impl Default for ProverConfig {
             max_concurrency_limit: default_max_concurrency_limit(),
             max_request_duration: default_max_request_duration(),
             max_buffered_queries: default_max_buffered_queries(),
-            cpu_prover: CpuProverConfig::default(),
-            network_prover: NetworkProverConfig::default(),
-            gpu_prover: GpuProverConfig::default(),
+            primary_prover: AgglayerProverType::NetworkProver(NetworkProverConfig::default()),
+            fallback_prover: None,
             grpc: Default::default(),
         }
     }
 }
 
+impl ProverConfig {
+    pub fn try_load(path: &Path) -> Result<Self, ConfigurationError> {
+        let reader = std::fs::read_to_string(path).map_err(|source| {
+            ConfigurationError::UnableToReadConfigFile {
+                path: path.to_path_buf(),
+                source,
+            }
+        })?;
+
+        let deserializer = toml::de::Deserializer::new(&reader);
+        serde::Deserialize::deserialize(deserializer)
+            .map_err(ConfigurationError::DeserializationError)
+    }
+}
+
+/// Type of the prover to be used for generation of the pessimistic proof
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub enum AgglayerProverType {
```

Comment:
Other config `enum`s use `#[serde(rename_all = "kebab-case")]`. See for example `agglayer_config::epoch::Epoch`. The config key names will then be `network-prover` instead of `NetworkProver` and such.